packages/grid/_modules_/grid/hooks/features/filter/useGridFilter.ts

```diff
@@ -118,6 +118,7 @@ export const useGridFilter = (
         return {
           ...state,
           visibleRows: {
+            ...state.visibleRows,
             visibleRowsLookup,
             visibleRows: Object.entries(visibleRowsLookup)
               .filter(([, isVisible]) => isVisible)
```

Comment:
The logic of the data grid is not consistent regarding this behavior. Sometimes, we spread extra props even if we override them all, sometimes, we don't. The change seems fair as a systematic approach. It will reduce the cost of future updates.